core/src/test/java/hudson/security/HudsonPrivateSecurityRealmTest.java

```diff
@@ -0,0 +1,134 @@
+package hudson.security;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.mockStatic;
+import static org.mockito.Mockito.when;
+
+import hudson.security.HudsonPrivateSecurityRealm.JBCryptEncoder;
+import hudson.security.HudsonPrivateSecurityRealm.PBKDF2PasswordEncoder;
+import java.security.NoSuchAlgorithmException;
+import java.security.spec.InvalidKeySpecException;
+import java.time.Duration;
+import javax.crypto.SecretKeyFactory;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+
+public class HudsonPrivateSecurityRealmTest {
+
+    // MySecurePassword
+    private static final String PBKDF2_HMAC_SHA512_ENCODED_PASSWORD =
+            "$HMACSHA512:210000:30f9e0a5470a8bc67f128ca1aae25dd4$88abaca4f442caeff0096ec0f75df2d77cc31a956c564133232f4d2532a72c8d4380a718d5b2a3dccab9e752027eeadd8f9f2c0c624505531bf3a57ec7d08aad";
+
+    //@Test
```

Comment:
```suggestion
    /* This exists so that we can easily  check the complexity of how long this takes (ie is the number of iterations we 
     * use correct for the state of CPUs).  
     * We do not want to assert that the range < x and  > y  as that would make the test flaky on overloaded
     *  or slow hardware, so this is commented out but left for ease of running locally when desired.
     */
    //@Test
```